modules/core/include/opencv2/core/detail/reduce_arg_helper.impl.hpp

```diff
@@ -0,0 +1,57 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_CORE_DETAIL_REDUCE_ARG_HELPER_IMPL_HPP
+#define OPENCV_CORE_DETAIL_REDUCE_ARG_HELPER_IMPL_HPP
+
+namespace cv {
+namespace detail {
+
+//! @cond IGNORED
+
+enum class ReduceMode
+{
+    FIRST_MIN = 0, //!< get index of first min occurrence
+    LAST_MIN  = 1, //!< get index of last min occurrence
+    FIRST_MAX = 2, //!< get index of first max occurrence
+    LAST_MAX  = 3, //!< get index of last max occurrence
+};
```

Comment:
> class ReduceMode

We don't need to expose these implementation details.

Tests may use own enum (or functor which is better).

---

I expected to put to includes:
- `depthDispatch()` only.
- use `dispatch_helper.impl.hpp` as a file name: https://github.com/opencv/opencv/pull/20733#discussion_r752386931